src/lazy/binary/raw/v1_1/immutable_buffer.rs

```diff
@@ -340,19 +335,114 @@ impl<'a> ImmutableBuffer<'a> {
+        match opcode.opcode_type {
+            OpcodeType::AnnotationFlexSym => self.read_flex_sym_annotations_sequence(opcode),
+            OpcodeType::SymbolAddress => self.read_symbol_address_annotations_sequence(opcode),
+            _ => unreachable!("read_annotations_sequence called for non-annotations opcode"),
+        }
+    }
+
+    fn read_flex_sym_annotations_sequence(
         self,
-        mut _type_descriptor: Opcode,
-    ) -> IonResult<LazyRawBinaryValue_1_1<'a>> {
-        todo!();
+        opcode: Opcode,
+    ) -> ParseResult<'a, EncodedAnnotations> {
+        let input_after_opcode = self.consume(1);
+        // TODO: This implementation actively reads the annotations, which isn't necessary.
+        //       At this phase of parsing we can just identify the buffer slice that contains
+        //       the annotations and remember their encoding; later on, the annotations iterator
+        //       can actually do the reading. That optimization would be impactful for FlexSyms
+        //       that represent inline text.
+        let (sequence, remaining_input) = match opcode.length_code {
+            7 => {
+                let (flex_sym, remaining_input) = input_after_opcode.read_flex_sym()?;
+                let sequence = EncodedAnnotations {
+                    encoding: AnnotationsEncoding::FlexSym,
+                    header_length: 1, // 0xE7
+                    sequence_length: u16::try_from(flex_sym.size_in_bytes()).map_err(|_| {
+                        IonError::decoding_error(
+                            "the maximum supported annotations sequence length is 65KB.",
+                        )
+                    })?,
+                };
+                (sequence, remaining_input)
+            }
+            8 => {
+                let (flex_sym1, input_after_sym1) = input_after_opcode.read_flex_sym()?;
+                let (flex_sym2, input_after_sym2) = input_after_sym1.read_flex_sym()?;
+                let combined_length = flex_sym1.size_in_bytes() + flex_sym2.size_in_bytes();
+                let sequence = EncodedAnnotations {
+                    encoding: AnnotationsEncoding::FlexSym,
+                    header_length: 1, // 0xE8
+                    sequence_length: u16::try_from(combined_length).map_err(|_| {
+                        IonError::decoding_error(
+                            "the maximum supported annotations sequence length is 65KB.",
+                        )
+                    })?,
+                };
+                (sequence, input_after_sym2)
+            }
+            9 => {
+                let (flex_uint, remaining_input) = input_after_opcode.read_flex_uint()?;
+                let sequence = EncodedAnnotations {
+                    encoding: AnnotationsEncoding::FlexSym,
+                    header_length: u8::try_from(1 + flex_uint.size_in_bytes()).map_err(|_| {
+                        IonError::decoding_error("found a 256+ byte annotations header")
```

Comment:
Can `FlexUInt` even read a number this large? I thought it was limited to 128 bits.

Reply:
It can't; the current implementation would always raise an error during the `FlexUInt` read. Would you like to see this be an `unreachable!` instead?

Reply:
I'm going to leave this as-is for the moment. If profiling shows that an `unreachable!` or a `cold_code!` would help out here I'll change it.